homeassistant/components/shelly/__init__.py

```diff
@@ -39,18 +45,36 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         entry.data.get(CONF_PASSWORD),
         temperature_unit,
     )
+
+    if COAP_CONTEXT not in hass.data[DOMAIN][DATA_GENERAL]:
+        coap_context = hass.data[DOMAIN][DATA_GENERAL][
+            COAP_CONTEXT
+        ] = await aiocoap.Context.create_client_context()
+    else:
+        coap_context = hass.data[DOMAIN][DATA_GENERAL][COAP_CONTEXT]
+
+    async def shutdown_listener(*_):
+        """Home Assistant shutdown listener."""
+        await coap_context.shutdown()
+
+    if UNDO_SHUTDOWN_LISTENER not in hass.data[DOMAIN][DATA_GENERAL]:
```

Comment:
Instead of checking this, just do this only when you have to create the context. That way you know it's there only once.

Reply:
OK, now `coap_context` and listener are created in `async_setup` and `async_listen_once` is used to create the listener.